armv7l images are unable to run nixos-rebuild, nix, etc.

## Issue description

Not sure this is the correct repo. Will refile if needed, just let me know.

The wiki at https://nixos.wiki/wiki/NixOS_on_ARM links to http://nixos-arm.dezgeg.me/installer maintained by @dezgeg , and the armv7l image is affected by https://github.com/NixOS/nix/issues/2393 , meaning that it gets wedged and you cannot upgrade.

(Aside: The aarch64 image has nix 2.1.1, which is unaffected, but the disk image does not match the checksum in SHA256SUMS.)

### Steps to reproduce

Download https://www.cs.helsinki.fi/u/tmtynkky/nixos-arm/installer/sd-image-armv7l-linux.img , `dd` it to a micro sd and stick it in a pi 3b. Most invocations of `nix` will hit something that wants an `xz` archive, and from there it's impossible to proceed due to the linked bug above.

### Workaround

It is possible (if you have a day or so) to get a bootstrap toolchain from outside nix, install it, build nix and all its dependencies by hand, and then try to use an external nix to fix the nixos install. I have attached my notes to this issue: [upgrade-nix.txt](https://github.com/NixOS/nixpkgs/files/2661012/upgrade-nix.txt)

Hm, I wonder if someone can help me. My machine finally finished its update! And now I am trying to make my own sd-image.

My problem is with `lkl`, which refuses to build on armv7. After adding `armv7l-linux` to its list of platforms, I get the build error:
```
lib/hijack/hijack.c: In function 'fcntl':
lib/hijack/hijack.c:242:18: error: '__lkl__NR_fcntl' undeclared (first use in this function); did you mean '__lkl__NR_fcntl64'?
  return lkl_call(__lkl__NR_fcntl, 3, fd, lkl_fcntl_cmd_xlate(cmd), arg);
                  ^~~~~~~~~~~~~~~
                  __lkl__NR_fcntl64
```
Which seems maybe related to https://github.com/lkl/linux/pull/372? Their fix seems to specifically look for android though, which won't help of course.

Any ideas?

Thinking differently, the package needs `lkl` only to make use of the `cptofs` command inside the builder. Is there an alternative way to get this effect? I guess we cannot just mount the fs and `cp` things to it in the build environment?

--- a/pkgs/applications/virtualization/lkl/default.nix
+++ b/pkgs/applications/virtualization/lkl/default.nix
@@ -2,8 +2,9 @@
 , buildPackages }:
 
 stdenv.mkDerivation rec {
-  name = "lkl-2018-08-22";
-  rev  = "5221c547af3d29582703f01049617a6bf9f6232a";
+  pname = "lkl";
+  version = "2019-06-20";
+  rev  = "0a4ebeadad12b94db665b8daf30e44e949a02d90";
 
   outputs = [ "dev" "lib" "out" ];
 
@@ -15,7 +16,7 @@ stdenv.mkDerivation rec {
     inherit rev;
     owner  = "lkl";
     repo   = "linux";
-    sha256 = "1k2plyx40xaphm8zsk2dd1lyv6dhsp7kj6hfmdgiamvl80bjajqy";
+    sha256 = "09y9ci3j0lp3z74h577ng2s9a4n4glqpqndyc9iy6wdqaj764gm6";
   };
 
   # Fix a /usr/bin/env reference in here that breaks sandboxed builds
@@ -58,7 +59,7 @@ stdenv.mkDerivation rec {
       overhead
     '';
     homepage    = https://github.com/lkl/linux/;
-    platforms   = [ "x86_64-linux" "aarch64-linux" ]; # Darwin probably works too but I haven't tested it
+    platforms   = [ "x86_64-linux" "aarch64-linux" "armv7l-linux" "armv6l-linux" ]; # Darwin probably works too but I haven't tested it
     license     = licenses.gpl2;
     maintainers = with maintainers; [ copumpkin ];
   };

